Xamarin.PropertyEditing.Mac/Controls/BindingEditor/BindingEditorWindow.cs

```diff
@@ -0,0 +1,274 @@
+				NSLayoutConstraint.Create (pathSelectorControl, NSLayoutAttribute.Height, NSLayoutRelation.Equal, this.PathBox, NSLayoutAttribute.Height, 1f, -10f)
+			});
+
+			this.viewModel.PropertyChanged += (sender, e) => {
+				if (e.PropertyName == nameof (CreateBindingViewModel.ShowLongDescription)) {
+					longDescription.Hidden = !this.viewModel.ShowLongDescription;
+					longDescription.StringValue = this.viewModel.ShowLongDescription ? this.viewModel.SelectedBindingSource.Description : string.Empty;
+					typeHeader.Hidden = this.viewModel.ShowLongDescription;
+				}
+
+				if (e.PropertyName == nameof (CreateBindingViewModel.ShowObjectSelector)) {
+					if (this.viewModel.ShowObjectSelector) {
+						typeHeader.Title = Properties.Resources.SelectObjectTitle;
+					}
+				}
+
+				if (e.PropertyName == nameof (CreateBindingViewModel.ShowTypeSelector)) {
+					typeSelectorControl.Hidden = !this.viewModel.ShowTypeSelector;
+
+					if (this.viewModel.ShowTypeSelector) {
+						typeHeader.Title = Properties.Resources.SelectTypeTitle;
+
+						if (this.viewModel.ShowTypeSelector && this.viewModel.TypeSelector != null) {
+							typeSelectorControl.ViewModel = this.viewModel.TypeSelector;
+						}
+					}
+				}
+
+				if (e.PropertyName == nameof (CreateBindingViewModel.ShowResourceSelector)) {
+					if (this.viewModel.ShowResourceSelector) {
+						typeHeader.Title = Properties.Resources.SelectResourceTitle;
+					}
+				}
+			};
+
+			this.viewModel.CreateValueConverterRequested += OnCreateValueConverterRequested;
+
+			this.ButtonDone.Activated += (sender, e) => {
+				if (pathSelectorControl.CustomPath.Enabled && !string.IsNullOrEmpty (pathSelectorControl.CustomPath.Cell.Title)) {
+					this.viewModel.Path = pathSelectorControl.CustomPath.Cell.Title;
+				}
+
+				Close ();
+			};
+
+			// More Settings
+			var controlTop = 6;
+			foreach (PropertyViewModel item in this.viewModel.BindingProperties) {
+
+				var label = new UnfocusableTextField {
+					StringValue = item.Property.Name,
+					TranslatesAutoresizingMaskIntoConstraints = false,
+				};
+				this.MoreSettingsView.AddSubview (label);
+
+				this.MoreSettingsView.AddConstraints (new[] {
+					NSLayoutConstraint.Create (label, NSLayoutAttribute.Top, NSLayoutRelation.Equal, this.MoreSettingsView, NSLayoutAttribute.Top, 1f, controlTop),
+					NSLayoutConstraint.Create (label, NSLayoutAttribute.Left, NSLayoutRelation.Equal, this.MoreSettingsView, NSLayoutAttribute.Left, 1f, 16f),
+					NSLayoutConstraint.Create (label, NSLayoutAttribute.Height, NSLayoutRelation.Equal, 1f, 18f),
+				});
+
+				var textField = new NSTextField {
+					Enabled = item.Property.CanWrite,
+					StringValue = item.Name,
+					TranslatesAutoresizingMaskIntoConstraints = false,
+				};
+				textField.Changed += (sender, e) => {
+					// TODO What do we update in the VM with this value?
```

Comment:
@ermau I'm finding it difficult to pin point exactly which property on the VM this should hook up to. Any pointers?

Reply:
It shouldn't even be a text field, it should be a full on property presenter.

Reply:
@ermau I don't have notes saying that section should have specifically been a property presenter, only that that section was dynamic based on the `BindingProperties` and `FlagProperties`. Could you clarify where/when that was mentioned, I must have missed that?

So were you thinking that we should re-use `EditorContainer` or did you have something else in mind or am I totally off track here?

Reply:
I don't know what else you would do, why would you expect to take random `PropertyViewModel`s from `BindingProperties` and make a new editor for them here instead of reusing everything we've already built?

Yes, we should reuse `EditorContainer`.